custom_model_runner/datarobot_drum/drum/model_metadata.py

```diff
@@ -0,0 +1,105 @@
+Released under the terms of DataRobot Tool and Utility Agreement.
+"""
+
+import trafaret as t
+
+
+from datarobot_drum.drum.enum import ModelMetadataHyperParamTypes
+
+
+# Max length of a user-defined parameter
+PARAM_NAME_MAX_LENGTH = 64
+
+# Max length of a select value
+PARAM_SELECT_VALUE_MAX_LENGTH = 32
+
+# Max number of possible select values
+PARAM_SELECT_NUM_VALUES_MAX_LENGTH = 24
+
+# Max length of the value of a string/unicode parameter
+PARAM_STRING_MAX_LENGTH = 1024
+
+
+IntHyperParameterTrafaret = t.Dict(
+    {
+        t.Key("name"): t.String(max_length=PARAM_NAME_MAX_LENGTH),
+        t.Key("type"): t.Enum("int"),
+        t.Key("min"): t.Int,
+        t.Key("max"): t.Int,
+        t.Key("default", optional=True): t.Int,
+    }
+)
+
+FloatHyperParameterTrafaret = t.Dict(
+    {
+        t.Key("name"): t.String(max_length=PARAM_NAME_MAX_LENGTH),
+        t.Key("type"): t.Enum("float"),
+        t.Key("min"): t.Float,
+        t.Key("max"): t.Float,
+        t.Key("default", optional=True): t.Float,
+    }
+)
+
+StringHyperParameterTrafaret = t.Dict(
+    {
+        t.Key("name"): t.String(max_length=PARAM_NAME_MAX_LENGTH),
+        t.Key("type"): t.Enum("string"),
+        t.Key("default", optional=True): t.String(
+            max_length=PARAM_STRING_MAX_LENGTH, allow_blank=True
+        ),
+    }
+)
+
+SelectHyperParameterTrafaret = t.Dict(
+    {
+        t.Key("name"): t.String(max_length=PARAM_NAME_MAX_LENGTH),
+        t.Key("type"): t.Enum("select"),
+        t.Key("values"): t.List(
+            t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH, allow_blank=False),
+            min_length=1,
+            max_length=PARAM_SELECT_NUM_VALUES_MAX_LENGTH,
+        ),
+        t.Key("default", optional=True): t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH),
+    }
+)
+
+# Multi only supports int, float, or select
+MultiHyperParameterTrafaret = t.Dict(
+    {
+        t.Key("name"): t.String(max_length=PARAM_NAME_MAX_LENGTH),
+        t.Key("type"): t.Enum("multi"),
+        t.Key("values"): t.Dict(
+            {
+                t.Key("int", optional=True): t.Dict({t.Key("min"): t.Int, t.Key("max"): t.Int,}),
+                t.Key("float", optional=True): t.Dict(
+                    {t.Key("min"): t.Float, t.Key("max"): t.Float,}
+                ),
+                t.Key("select", optional=True): t.Dict(
+                    {
+                        t.Key("values"): t.List(
+                            t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH, allow_blank=False),
+                            min_length=1,
+                            max_length=PARAM_SELECT_NUM_VALUES_MAX_LENGTH,
+                        ),
+                    }
+                ),
+            }
+        ),
+        t.Key("default", optional=True): t.Or(
+            t.Int, t.Float, t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH)
+        ),
+    }
+)
+
+
+HyperParameterTrafaret = {
```

Comment:
It was not implemented as t.List(t.Or(...)). `t.Or()` will run all the validators and return error message of each validator if the validation fails. The error message is difficult to parse and read by users.